agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -150,147 +151,146 @@ private static String extractArtefactUrl(URL jarOrClassFolderUrl) {
-	}
-
-	private static boolean isNestedInWar(String filePath) {
-		return filePath.contains(WAR_FILE_ENDING) && filePath.endsWith(JAR_FILE_ENDING);
-	}
-
-	private static boolean isNestedInFatJar(String filePath) {
-		return filePath.contains(JAR_FILE_ENDING) &&
-				filePath.indexOf(JAR_FILE_ENDING) != filePath.length() - JAR_FILE_ENDING.length();
+		return findGitPropertiesInDirectoryFile(file, recursiveSearch);
 	}
 
-	private static Pair<String, Properties> findGitPropertiesInArchiveFile(File file) throws IOException {
+	/**
+	 * Searches for git properties in jar/war/ear/aar files
+	 */
+	private static List<Pair<String, Properties>> findGitPropertiesInArchiveFile(File file,
+																				 boolean recursiveSearch) throws IOException {
 		try (JarInputStream jarStream = new JarInputStream(
-				new BashFileSkippingInputStream(new FileInputStream(file)))) {
-			return findGitPropertiesInArchive(jarStream);
+				new BashFileSkippingInputStream(Files.newInputStream(file.toPath())))) {
+			return findGitPropertiesInArchive(jarStream, file.getName(), recursiveSearch);
 		} catch (IOException e) {
 			throw new IOException("Reading jar " + file.getAbsolutePath() + " for obtaining commit " +
 					"descriptor from git.properties failed", e);
 		}
 	}
 
-	/** Searches for a git.properties file inside a jar file that is nested inside a jar or war file. */
-	static Pair<String, Properties> findGitPropertiesInNestedArchiveFile(File file) throws IOException {
-		String filePath = file.getPath();
-		int firstPartEndIndex;
-		if (filePath.contains(WAR_FILE_ENDING)) {
-			firstPartEndIndex = filePath.indexOf(WAR_FILE_ENDING) + WAR_FILE_ENDING.length();
-		} else {
-			firstPartEndIndex = filePath.indexOf(JAR_FILE_ENDING) + JAR_FILE_ENDING.length();
-		}
-		String firstPart = filePath.substring(0, firstPartEndIndex);
-		String fileName = file.getName();
-		try (JarInputStream jarStream = new JarInputStream(
-				new BashFileSkippingInputStream(new FileInputStream(firstPart)))) {
-			Pair<String, JarInputStream> nestedJar = findEntry(jarStream, fileName);
-			if (nestedJar == null) {
-				return null;
-			}
-			JarInputStream nestedJarStream = new JarInputStream(nestedJar.getSecond());
-			return findGitPropertiesInArchive(nestedJarStream);
+	/**
+	 * Searches for git.properties file in the given folder
+	 *
+	 * @param recursiveSearch If enabled, git.properties files will also be searched in jar files
+	 */
+	private static List<Pair<String, Properties>> findGitPropertiesInDirectoryFile(
+			File directoryFile, boolean recursiveSearch) throws IOException {
+		List<Pair<String, Properties>> result = new ArrayList<>();
+		result.addAll(findGitPropertiesInFolder(directoryFile));
 
-		} catch (IOException e) {
-			throw new IOException("Reading jar " + firstPart + " for obtaining commit " +
-					"descriptor from git.properties failed", e);
+		if (recursiveSearch) {
+			result.addAll(findInJarFilesNestedGitPropertiesFilesInFolder(directoryFile));
```

Comment:
Maybe you could split this function into `findJarFilesInFolder` and `findGitPropertiesInJarFiles` or something like that. The name makes me think that this function has too many responsibilities 😄

Reply:
Hm, yes and no. I renamed it, hopefully that makes things clearer.
I implemented it in a way so that `findGitPropertiesInDirectoryFile` is clearly understandable and we don't get more small utility functions with vaguely the same name (findXinY) It's confusing at it is already 😉